lib/game_logic.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+class Game
+  attr_reader :turn_counter
+  def initialize
+    @board = { a1: "\s", a2: "\s", a3: "\s", b1: "\s", b2: "\s", b3: "\s", c1: "\s", c2: "\s", c3: "\s" }
+    @turn_counter = 1
+    @players = []
+    @turns = 9
+  end
+
+  def turn_incrementor
+    @turn_counter += 1
+  end
+
+  def turn_reseter
+    @turn_counter = 1
+  end
+
+  def game_loop_on
+    @turn_counter <= @turns
+  end
+
+  def create_player(name)
+    player = Player.new(name)
+    @players << player
+    player
+  end
+
+  def display_board
+    "    1  2  3\n a [#{@board[:a1]}][#{@board[:a2]}][#{@board[:a3]}]
+ b [#{@board[:b1]}][#{@board[:b2]}][#{@board[:b3]}]
+ c [#{@board[:c1]}][#{@board[:c2]}][#{@board[:c3]}]\n\n"
+  end
+
+  def board_values
+    @board.values.join
+  end
+
```

Comment:
**Optional**  It will be better to put board related methods in a `Board  class`

Reply:
Done. Great suggestion!